### ISSUE TITLE
x/crypto/pkcs12: unknown attribute with OID 1.3.6.1.4.1.311.17.3.92

### ISSUE THREAD

**Issue by leeview** [golang/go#24325]
Please answer these questions before submitting your issue. Thanks!

### What version of Go are you using (`go version`)?
1.10

### Does this issue reproduce with the latest release?
yes

### What operating system and processor architecture are you using (`go env`)?
windows 10 amd64

### What did you do?
I was converting a PFX file to pem.Block using the golang.org/x/crypto/pkcs12.ToPEM() function

### What did you expect to see?

the certificate and the key PEM bytes

### What did you see instead?

error: pkcs12: unknown attribute with OID 1.3.6.1.4.1.311.17.3.92

http://oid-info.com/get/1.3.6.1.4.1.311.17.3.92

Apparently PFX files (on windows) can contain an OID which is not handled by pkcs12 parser.

https://github.com/golang/crypto/blob/master/pkcs12/pkcs12.go#L177

Is there a way to handle this OID or just ignore it if it's not relevant to the output ?
Thank you

**Comment by sooryaprakash99** [comment]
I am also getting the above error with go lang version 1.8.7 .  Is there any solution

**Comment by adamdecaf** [comment]
It looks like we would need to decode the [blob they send](https://msdn.microsoft.com/en-us/7a06eae5-96d8-4ece-98cb-cf0710d2ddbd) (`CRYPT_INTEGER_BLOB `). 

Docs: https://docs.microsoft.com/en-us/windows/desktop/api/wincrypt/nf-wincrypt-certsetcertificatecontextproperty#cert_subject_pub_key_bit_length_prop_id

FWIW I don't see that specific OID on their page, but are there lots of others we'd need to worry about? 

https://support.microsoft.com/en-us/help/287547/object-ids-associated-with-microsoft-cryptography

**Comment by andrewstuart** [comment]
Interestingly, a very similar issue has cropped up in our environment just yesterday (for OID [`1 3 6 1 4 1 311 17 2`](https://github.com/openssl/openssl/blob/master/crypto/objects/objects.txt#L344). Not sure if there was a Microsoft update or what that recently caused us to start seeing this issue, but we've also had a really hard time tracking down documentation on this particular OID. In our case, the `attribute.Value.Bytes` is a bunch of `\0` bytes, so we've created a mirror as a temporary workaround that ignores empty (after bytes.Trim) OIDs.

It's really hard to figure out what to do with the bytes in our case if they ever are not empty (though honestly in my case, we don't care, we just want the cert, but in the general case that may not be true depending on what they mean).

The best I've been able to come up with as an action plan is to dive into OpenSSL and see how they're handling it, but I'm pretty unfamiliar with both the language and codebase.
